packages/react/src/utils.ts

```diff
@@ -0,0 +1,173 @@
+import { GadgetError, gadgetErrorFor, getNonNullableError, InvalidFieldError } from "@gadgetinc/api-client-core";
+import type { CombinedError, OperationResult } from "@urql/core";
+import { GraphQLError } from "graphql";
+import { Operation, OperationContext, UseQueryState } from "urql";
+
+/**
+ * The inner result object returned from a query result
+ **/
+export interface ReadHookState<Data = any, Variables = Record<string, any>> {
+  fetching: boolean;
+  stale: boolean;
+  data?: Data;
+  error?: ErrorWrapper;
+  extensions?: Record<string, any>;
+  operation?: Operation<Data, Variables>;
+}
```

Comment:
It looks like your hands are tied for `ReadHookResult`, but for this interface, what do you think of just extending the `urql` type instead, to avoid hardcoding type defs for props you're not actually decorating here? Like so:
```suggestion
export interface ReadHookState<Data = any, Variables = Record<string, any>> extends UseQueryState<Data, Variables> {
  error?: ErrorWrapper
}
```

Reply:
I think that is cleaner from a reuse perspective but I find that these type definitions are in a way user facing code, in that when folks hit go to type definition on a thing, they get taken to this text exactly. For that reason I kinda felt like hardcoding it was nice to show it all in one place as opposed to asking the user to understand how we subtype and override urqls types, you know?

Reply:
Yeah, fair point. I don't imagine this would be at a high risk of changing, at least on a minor version.